pkg/update/deps/cleanup.go

```diff
@@ -30,7 +30,7 @@ func gitCheckout(p *config.Pipeline, dir string, mutations map[string]string) er
 	}
 
 	tagValue := p.With["tag"]
-	if tagValue == "" {
+	if tagValue == "" && p.With["branch"] == "" {
```

Comment:
nit: I might extract the expression `p.With["branch"]` into a value since we end up accessing it 3 times here